src/main/java/mekanism/common/content/gear/mekatool/ModuleBlastingUnit.java

```diff
@@ -0,0 +1,110 @@
+package mekanism.common.content.gear.mekatool;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Consumer;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import mekanism.api.gear.ICustomModule;
+import mekanism.api.gear.IModule;
+import mekanism.api.gear.config.IModuleConfigItem;
+import mekanism.api.gear.config.ModuleConfigItemCreator;
+import mekanism.api.gear.config.ModuleEnumData;
+import mekanism.api.text.EnumColor;
+import mekanism.api.text.IHasTextComponent;
+import mekanism.api.text.TextComponentUtil;
+import mekanism.common.MekanismLang;
+import mekanism.common.block.BlockBounding;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockState;
+
+@ParametersAreNonnullByDefault
+public class ModuleBlastingUnit implements ICustomModule<ModuleBlastingUnit> {
+
+    private IModuleConfigItem<BlastRadius> blastRadius;
+
+    @Override
+    public void init(IModule<ModuleBlastingUnit> module, ModuleConfigItemCreator configItemCreator) {
+        blastRadius = configItemCreator.createConfigItem("blast_radius", MekanismLang.MODULE_BLAST_RADIUS,
+                new ModuleEnumData<>(BlastRadius.class, module.getInstalledCount() + 1, BlastRadius.LOW));
+    }
+
+    public int getBlastRadius() {
+        return blastRadius.get().getRadius();
+    }
+
+    public static Set<BlockPos> findPositions(Level world, BlockPos location, Direction axis, int radius) {
+        int minX = 0, minY = 0, minZ = 0;
+        int maxX = 0, maxY = 0, maxZ = 0;
+        switch (axis) {
+            case UP, DOWN -> {
+                minX = -radius;
+                maxX = radius;
+                minY = maxY = 0;
+                minZ = -radius;
+                maxZ = radius;
+            }
+            case EAST, WEST -> {
+                minX = maxX = 0;
+                minY = (radius == 0) ? 0 : -1;
+                maxY = (radius == 0) ? 0 : (2 * radius - 1);
+                minZ = -radius;
+                maxZ = radius;
+            }
+            case NORTH, SOUTH -> {
+                minX = -radius;
+                maxX = radius;
+                minY = (radius == 0) ? 0 : -1;
+                maxY = (radius == 0) ? 0 : (2 * radius - 1);
+                minZ = maxZ = 0;
+            }
+        }
+        Set<BlockPos> found = new HashSet<>();
+        for (BlockPos pos : BlockPos.betweenClosed(location.offset(minX, minY, minZ), location.offset(maxX, maxY, maxZ))) {
+            BlockState blockState = world.getBlockState(pos);
+            if (!blockState.isAir() && !(blockState.getBlock() instanceof BlockBounding) && !(blockState.destroySpeed < 0)) {
```

Comment:
At the very least this should be checking the destroy speed method rather than the constant in case anything overrides it `blockState#getDestroySpeed(world, pos)` but part of me also wonders if instead it should be using `Blockstate#getDestroyProgress` so that it can properly calculate the player relative hardness unless that is already being handled elsewhere.

Reply:
So I don't understand the change to getDestroyProgress, this is a check for unbreakable blocks which I think I copied from the digital miner, but maybe it was using the method.
Also what are your thoughts on only area mining blocks with hardness less than or equal to the target block?

Reply:
Yeah the digital miner uses the method for checking the destroy speed. The reason I was suggesting the player specific one (though it may not matter due to checks that happen when trying to actually break the block) was that unlike for the digital miner we actually have a player context for who is trying to break the blocks. And I think it would make sense to only area mine blocks that are of a comparable hardness so that you can't just say place a piece of dirt next to obsidian and then break the dirt to instant mine the obsidian.

Reply:
So I think there's three reasonable ways to go about the hardness thing:
1. Use the hardness of the hardest block you're attempting to mine
2. Only mine the blocks with hardness <= hardness of target. (Basically 1 but with the target defined as the hardest block)
3. Add all of the hardness's you're attempting to mine, kinda just mining in bulk really.
I think I might like 2 best and it's definitely the easiest to implement, but honestly 1 would be the nicest to use.

Reply:
Yeah honestly best bet is probably to go with option 2 as that will be easiest to implement but the *nicest* would be semi similar to number one where it would be a weighted average based on the hardest you are attempting to mine (or maybe as you say just use the hardest overall as is)

Reply:
I looked at the getDestroyProgress and I still like just using getDestroySpeed. But you might be right that mods could be using the getDestroyProgress to make things unbreakable by setting the dig speed on the event (ForgeEventFactory.getBreakSpeed) that it calls to 0. If we change it here though we should probably add that to the vein mining as it might run into the same situation. For now I'm just gonna change to getDestroySpeed, just let me know if you think I should change that.

Reply:
I tried with 2, I hate it... it's still useful, but grass being harder than dirt is gonna get really annoying, also mining stone won't get any of the ores which makes vein mining semi useless to use with this too. I left it on 2 for now, but I'm gonna have to look into doing some version of 1 I think.

Reply:
So I don't really like calling findPositions every frame for the outline, and am hesitant to be now calling it every tick for option especially since it'll be calling getDestroyProgress on each of the positions to find the lowest if we're going to be going with the hardest or an average. Is there anyway we can be caching it? Because of possible changes in the vicinity I'm thinking maybe not, but if there was it would make me feel much more confident about performance.

Reply:
In theory I suppose if we track which position is the hardest instead of just how hard we could then do:
```java
event.setNewSpeed(player.getDigSpeed(hardestState, hardestPos))
```

Though similar to the blasting rendering we would need a marker boolean to ensure we don't try to modify the dig speed of the other position.

Reply:
I ended up cutting out hardness 0 blocks from area mining entirely, as far as the speed thing I think what we currently have works fine with the hardness ratio. If we were going to try tracking stuff we'd have to make sure that we handled if someone broke our hardest block at some point, and as this is server stuff I don't know if it's safe to do boolean flags if other players might be setting them.